### PR TITLE
Add app studio roles

### PR DESCRIPTION
**Pull request by ranakan19**
fix wrt: https://issues.redhat.com/browse/ASC-300

Add appstudio roles 
e2e test: https://github.com/codeready-toolchain/toolchain-e2e/pull/695

### CODE REVIEW

File: deploy/templates/nstemplatetiers/appstudio/spacerole_admin.yaml [hunk 1]
```diff
@@ -80,74 +83,19 @@ objects:
...
-    - get
-    - list
-    - watch
-    - update
-    - patch
-    - delete
-  - apiGroups:
-    - appstudio.redhat.com
-    resources:
-    - spiaccesstokendataupdates
```
Comment (alexeykazakov) [inline deploy/templates/nstemplatetiers/appstudio/spacerole_admin.yaml:93]: @skabashnyuk Can you confirm that we don't need users have access to `spiaccesstokendataupdates` resources? They are not listed in https://github.com/redhat-appstudio/book/blob/main/ADR/0011-roles-and-permissions.md

**Review by alexeykazakov — COMMENTED** [review]
Looks good! I just want to confirm with @skabashnyuk before finally approving the PR.

**Review by rajivnathan — COMMENTED** [review]
Are there any SpaceBindings on any of the clusters with viewer role? If so then we probably 
shouldn't remove the viewer role in this PR and plan migration for those SpaceBindings first?

File: deploy/templates/nstemplatetiers/appstudio/spacerole_admin.yaml (hunk 1 above)
Comment (ranakan19) [reply deploy/templates/nstemplatetiers/appstudio/spacerole_admin.yaml:93]: @alexeykazakov done in [dda2100](https://github.com/codeready-toolchain/host-operator/pull/775/commits/dda21007d5e2d5209ddd459960eb945d91fbc4e1)

**Review by alexeykazakov — APPROVED** [review]

File: deploy/templates/nstemplatetiers/appstudio-env/spacerole_contributor.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  name: appstudio-env-spacerole-maintainer
```
Comment (alexeykazakov) [inline deploy/templates/nstemplatetiers/appstudio-env/spacerole_contributor.yaml:4]:
```suggestion
  name: appstudio-env-spacerole-contributor
```

**Pull request merged**